src/plugins/intel_gpu/src/runtime/engine.cpp

```diff
@@ -217,4 +217,18 @@ std::shared_ptr<cldnn::engine> engine::create(engine_types engine_type,
     return engine::create(engine_type, runtime_type, default_device, configuration, task_executor);
 }
 
+memory_ptr engine::get_variable_memory(const std::string& variable_id, const layout& layout) {
```

Comment:
I decided to put variables in engine just for simplicity, since variables should be available both at program compilation stage and in runtime. Though from architecture point of view network indeed is more suitable place for variables.

Moved variables to `intel_gpu::Program` and `cldnn::network` classes.